File: software/backend/src/main/java/ru/scms/back/dto/StackDto.java
package ru.scms.back.dto;

import java.time.LocalDateTime;
import ru.scms.back.enums.StackStatus;

public record StackDto(
        Long id,
        String code,
        Long ownerUserId,
        String ownerName,
        StackStatus status,
        String location,
        LocalDateTime lastExtractedAt) {}

File: software/client-portal/src/shared/api/dto/stackDto.ts
import type { StackStatus } from "./enums";

export interface StackDto {
  id: number;
  code: string;
  ownerUserId: number | null;
  ownerName: string | null;
  status: StackStatus;
  location: string | null;
  lastExtractedAt: string | null;
}
